### ISSUE TITLE
Properties context menu generation improvements

### ISSUE THREAD

**Issue by ferdnyc** [OpenShot/openshot-qt#3783]
@jonoomph asked via email about the menu-locking functionality in the Properties context menu. It doesn't work particularly well. As he noted, "it seems to be blocking every other call to the context menu (on the properties table view)"

I responded:

> Yeah, I've noticed that it's not working perfectly, I keep meaning to look at that.
> 
> The original purpose (the PR I added it in is #2997: Properties: Ignore overlapping context menu events) was to address bug #2996, an issue with the context menus for things like the Source property of Mask transitions, and other properties that listed project files as selections. As I wrote in the PR:
> 
> > Because "reader" type properties require a fairly lengthy context menu that can take some time to display, this PR adds a lock variable to protect against repeated invocations of the event handler from being queued up during menu construction, and subsequently trampling all over each other.
> 
> Basically, if you right-clicked on a Source property, the menu takes some hundreds of milliseconds or even multiple seconds to actually come up. Giving plenty of opportunity to right-click again during that period, and then things would get totally confused. So, the lock was intended as a way of debouncing the context-menu request, by having the event handler ignore additional events if the menu is already in the process of being constructed. But, it seems not to limit itself to those situations as well as I'd hoped.
>  
> I think part of the solution might be to actually use a persistent single-shot timer, rather than the one-off ones I had been using, and to have that call a different slot to release the lock rather than relying on re-invoking the event handler (which seems not to work very well). My original idea was that, by re-using the handler, the unlocking would get priority over any other events waiting in the queue. But I think it really needs to be a separate signal, instead.
> 
> The other thing it could do is only set the lock when the request is for a reader-type property, since the other menu types come up quickly enough that they don't really need to be debounced.

Then, after thinking about it some more:

> Hm. Another option altogether, come to think of it — I added that locking (shortly) before we re-did all of the data models to make things persistent instead of constantly blowing away and re-creating them. The context menus still work the same way the old models used to do, they build themselves completely from scratch every time. That's why it takes so long for the Source context menu to build. It's also why we're actually bleeding menu objects, something I recently noticed in GammaRay. Things aren't getting cleaned up from all those single-use menu objects we build.
> 
> Now that the data models for the files and transitions and the like are more persistent, we could probably build a "property source proxy model" that combines both of those models together and provides the available source items in a form that's ready-made for the context menu to consume (icons already generated and etc.) Done that way, it might be quick enough to display that we wouldn't need to worry about debouncing at all.

**Comment by jonoomph** [comment]
Good point. I vote for removing the lock, because it seems to add complexity where none is needed (unless we are forgetting something)

**Comment by ferdnyc** [comment]
I have a version of the code in a project branch that removes the lock, it seems fine to me. I streamlined how the menus are generated and made sure they're all parented correctly, which seems to help as well. I'll file a PR as soon as I get confirmation whether some of the other changes I made improved some input weirdness @USATechDude reported on macOS.

I also fixed the ages-old bug where hitting the Rewind button while the preview is paused would accidentally make the player start moving **forward** instead of in reverse. 

And I added a simple textual progress display in the status bar, for imports of more than 15 files at a time. ("Importing M / N" -- localized, ofc.) I should really just replace it with an actual progress bar, like the new Blender one. But it serves it's main purpose, which is to keep the UI from freezing during the import.
